src/simplify_rules.jl

```diff
@@ -133,18 +159,35 @@ let
     global serial_expand_simplifier
 
     function default_simplifier(; kw...)
-        IfElse(has_trig_exp,
-               Postwalk(IfElse(x->symtype(x) <: Number,
-                               Chain((number_simplifier(),
-                                      trig_exp_simplifier())),
-                               If(x->symtype(x) <: Bool,
-                                  bool_simplifier()))
-                        ; kw...),
-               Postwalk(Chain((If(x->symtype(x) <: Number,
-                                  number_simplifier()),
-                               If(x->symtype(x) <: Bool,
-                                  bool_simplifier())))
-                        ; kw...))
+        IfElse(
+            has_trig_exp,
+            Postwalk(
+                IfElse(
+                    x->symtype(x) <: Number,
+                    Chain((number_simplifier(), trig_exp_simplifier())),
+                    If(x->symtype(x) <: Bool, bool_simplifier())
+                )
+                ; kw...
+            ),
+            IfElse(
+                has_div_rem_mod,
+                Postwalk(
+                    IfElse(
+                        x->symtype(x) <: Number,
+                        Chain((number_simplifier(), divremmod_simplifier())),
+                        If(x->symtype(x) <: Bool,bool_simplifier())
+                    )
+                    ; kw...
+                ),            
+                Postwalk(
+                    Chain(
+                        (If(x->symtype(x) <: Number, number_simplifier()),
+                        If(x->symtype(x) <: Bool, bool_simplifier()))
+                    )
+                    ; kw...
+                )
+            )
+        )
```

Comment:
This does not work when both predicates are true. How about something like:

```julia
chained(xs...) =  Postwalk(
                IfElse(
                    x->symtype(x) <: Number,
                    Chain(xs),
                    If(x->symtype(x) <: Bool, bool_simplifier())
                )
                ; kw...
            )
IfElse(has_trig_exp,
          IfElse(has_div_rem_mod,
               chained(number_simplifier(),  trig_exp_simplifier(), div_rem_mod_simplifier()),
               chained(number_simplifier(),  trig_exp_simplifier())),
          chained(number_simplifier()))
```

Reply:
This should fail if there is only div/rem/mod. But the use of `chained` should help. I will try something.

Reply:
It turns out the solution is much simpler:
```julia
    default_simplifier(; kw...) = Postwalk(
        Chain(
            (
                If(x->symtype(x) <: Number, number_simplifier()),
                If(has_trig_exp, trig_exp_simplifier()),
                If(has_div_rem_mod, div_rem_mod_simplifier()),
                If(x->symtype(x) <: Bool, bool_simplifier())
            )
        )
        ; kw...
    )
```

All test are good.

Reply:
It even seemed to have improved the performance, although I am not sure what I got with a single comparison is representative.